pkg/app/piped/cloudprovider/cloudrun/client.go

```diff
@@ -62,27 +62,103 @@ func newClient(ctx context.Context, projectID, region, credentialsFile string, l
 	return c, nil
 }
 
-func (c *client) Apply(ctx context.Context, sm ServiceManifest) (*Service, error) {
-	service, err := manifestToRunService(sm)
+func (c *client) Create(ctx context.Context, sm ServiceManifest) (*Service, error) {
+	svcCfg, err := manifestToRunService(sm)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		svc    = run.NewNamespacesServicesService(c.client)
+		parent = makeCloudRunParent(c.projectID)
+		call   = svc.Create(parent, svcCfg)
+	)
+	call.Context(ctx)
+
+	service, err := call.Do()
+	if err != nil {
+		if e, ok := err.(*googleapi.Error); ok && e.Code == http.StatusNotFound {
+			return nil, fmt.Errorf("failed to create service: code=%d, message=%s, details=%s", e.Code, e.Message, e.Details)
+		}
+		return nil, err
+	}
+	return (*Service)(service), nil
+}
+
+func (c *client) Update(ctx context.Context, sm ServiceManifest) (*Service, error) {
+	svcCfg, err := manifestToRunService(sm)
 	if err != nil {
 		return nil, err
 	}
 
 	var (
 		svc  = run.NewNamespacesServicesService(c.client)
 		name = makeCloudRunServiceName(c.projectID, sm.Name)
-		call = svc.ReplaceService(name, service)
+		call = svc.ReplaceService(name, svcCfg)
 	)
 	call.Context(ctx)
-	updatedService, err := call.Do()
+
+	service, err := call.Do()
 	if err != nil {
 		if e, ok := err.(*googleapi.Error); ok && e.Code == http.StatusNotFound {
-			return nil, fmt.Errorf("service %s was not found (%w), the service must be registered from Google CloudRun page", name, ErrServiceNotFound)
+			return nil, ErrServiceNotFound
 		}
 		return nil, err
 	}
+	return (*Service)(service), nil
+}
 
-	return (*Service)(updatedService), nil
+func (c *client) Apply(ctx context.Context, sm ServiceManifest) (*Service, error) {
+	svcCfg, err := manifestToRunService(sm)
+	if err != nil {
+		return nil, err
+	}
+
+	var notFound bool
+	update := func() (*Service, error) {
+		var (
+			svc  = run.NewNamespacesServicesService(c.client)
+			name = makeCloudRunServiceName(c.projectID, sm.Name)
+			call = svc.ReplaceService(name, svcCfg)
+		)
+		call.Context(ctx)
+
+		service, err := call.Do()
+		if err != nil {
+			if e, ok := err.(*googleapi.Error); ok && e.Code == http.StatusNotFound {
+				notFound = true
+				return nil, fmt.Errorf("service %s was not found (%w), the service must be registered from Google CloudRun page", name, ErrServiceNotFound)
```

Comment:
nit: this not found error message is eventually discarded. So how about making `errNotFound`
```
var errNotFound = errors.New("not found")
```

Then return it here
```suggestion
				return nil, errNotFound
```

Then check with `errors.Is(err, errNotFound)` at L142

Reply:
Opps. Forgot to remote this `Apply` function. It is no longer needed.

Reply:
Done!

Reply:
Lol, just got confused about this too 😂

Reply:
makes sense 😭